### DISCUSSION TITLE
Why my PrismaOptimize always reports `Connect Timeout Error`?

### DISCUSSION THREAD

**Discussion by jcyuan** [prisma/prisma#25738] [Q&A]
### Question

next.js project
@prisma/client: 5.22.0
@prisma/extension-optimize: 1.0.3

init code:
```typescript
import { PrismaClient } from '@prisma/client';
import { withOptimize } from "@prisma/extension-optimize";

const globalForPrisma = globalThis as unknown as { prisma: ReturnType<typeof createPrismaClient> };

function createPrismaClient() {
  return new PrismaClient({
    datasources: {
      db: {
        url: `${process.env.DATABASE_URL}`,
      },
    },
    log: [
      'warn',
      'error',
      {
        level: 'query',
        emit: process.env.NODE_ENV === 'production' ? 'event' : 'stdout',
      },
    ],
  }).$extends(withOptimize({ apiKey: process.env.OPTIMIZE_API_KEY }));
}

export const prisma = globalForPrisma.prisma ?? createPrismaClient();
if (process.env.NODE_ENV !== 'production')
  globalForPrisma.prisma = prisma;

export default prisma;
```

### How to reproduce (optional)

do:
```
prisma migrate reset
```
i have set 'seed' in my package.json, so this actually executes data inserting by the seed.ts at the last step, i guess seeding action will be captured by the optimizer to submit to prisma dashboard, but somehow it failed to submit data?
because i always got this error after seeding:

```
Running seed command `npx ts-node --compiler-options {"module":"CommonJS"} ./prisma/seed.ts` ...
┌─────────────────────────────────┐
│ See your Optimize dashboard at: │
│ https://optimize.prisma.io      │
└─────────────────────────────────┘
Seeding database...
Database seeded successfully!
TypeError: fetch failed
    at node:internal/deps/undici/undici:12502:13
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async E:\saas-template\node_modules\@prisma\extension-optimize\dist\index.js:1:957
    at async A (E:\saas-template\node_modules\@prisma\extension-optimize\dist\index.js:1:2067) {
  **[cause]: ConnectTimeoutError: Connect Timeout Error (attempted addresses: 157.240.0.18:443, 2001::1f0d:5f23:443)**
      at onConnectTimeout (node:internal/deps/undici/undici:6635:28)
      at node:internal/deps/undici/undici:6587:50
      at Immediate._onImmediate (node:internal/deps/undici/undici:6617:37)
      at processImmediate (node:internal/timers:478:21)
      at process.topLevelDomainCallback (node:domain:160:15)
      at process.callbackTrampoline (node:internal/async_hooks:128:24) {
    code: 'UND_ERR_CONNECT_TIMEOUT'
  }
}

An error occurred while running the seed command:
Error: Command failed with exit code 1: npx ts-node --compiler-options {"module":"CommonJS"} ./prisma/seed.ts
```

i have no idea what to do, how to fix this please?

thank you~

**Comment by ludralph** [comment]
Hi @jcyuan 

Thank you for raising this. I attempted to reproduce this issue but cannot reproduce this connect timeout error. Can you attempt the operation on another network to rule out network-specific restrictions? If you're using a VPN, you could maybe disable it to see if it works, or try using a VPN if you're not. This is to rule out any network issues on your end.

**Reply by jcyuan** [reply]
> Hi @jcyuan
> 
> Thank you for raising this. I attempted to reproduce this issue but cannot reproduce this connect timeout error. Can you attempt the operation on another network to rule out network-specific restrictions? If you're using a VPN, you could maybe disable it to see if it works, or try using a VPN if you're not. This is to rule out any network issues on your end.

thanks for your reply, i have tested pinging `157.240.0.18` and it didn't work, the Optimizer uses `undici` as the connection client, and i think it's impossible to set a proxy for it unless the source code is modified.
so i'm not sure how to test it with a local proxy.

Perhaps the prisma team could consider this senario?

**Reply by ludralph** [reply]
Hi @jcyuan 

Can you try setting the `HTTP_PROXY` or `HTTPS_PROXY` environment variables as described in the [documentation](https://www.prisma.io/docs/orm/reference/environment-variables-reference#proxy-environment-variables). This might affect the overall network behavior of your application:
```
export HTTP_PROXY=http://proxy.example.com
export HTTPS_PROXY=https://proxy.example.com
```
Replace `proxy.example.com` with your actual proxy server address

**Reply by jcyuan** [reply]
just tired, not work.

the doc says:
```
The Prisma CLI supports custom HTTP(S) proxies to download the Prisma engines. These can be helpful to use when working behind a corporate firewall. See [Using a HTTP proxy for the CLI](https://www.prisma.io/docs/orm/reference/prisma-cli-reference#using-a-http-proxy-for-the-cli) for more information.
```
that proxy setting is for downloading prisma engine, so i think it won't be applied for `undici`, and i have googled about `how to set proxy for undici`, no luck, all are coding to set.

**Reply by ludralph** [reply]
Hi @jcyuan 

Do you still experience this issue with the latest Prisma version 6.2.1?

**Reply by jcyuan** [reply]
> Hi @jcyuan
> 
> Do you still experience this issue with the latest Prisma version 6.2.1?

sorry for late reply, i need to upgrade Prisma? i think update of `@prisma/extension-optimize` is enough?

but after update to latest `^1.1.4`, still the same error:

```
TypeError: fetch failed
    at node:internal/deps/undici/undici:12502:13
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async eval (webpack-internal:///(rsc)/./node_modules/@prisma/extension-optimize/dist/index.js:1:12256)
    at async Ze (webpack-internal:///(rsc)/./node_modules/@prisma/extension-optimize/dist/index.js:1:14144)
    at async Q.onFlush (webpack-internal:///(rsc)/./node_modules/@prisma/extension-optimize/dist/index.js:1:18725)
    at async Q.forceFlush (webpack-internal:///(rsc)/./node_modules/@prisma/extension-optimize/dist/index.js:1:15626) {
  [cause]: ConnectTimeoutError: Connect Timeout Error (attempted addresses: 2a03:2880:f12a:83:face:b00c:0:25de:443, 157.240.18.18:443)
      at onConnectTimeout (node:internal/deps/undici/undici:6635:28)
      at node:internal/deps/undici/undici:6587:50
      at Immediate._onImmediate (node:internal/deps/undici/undici:6617:37)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'UND_ERR_CONNECT_TIMEOUT'
  }
}
 ⨯ unhandledRejection: TypeError: fetch failed
    at node:internal/deps/undici/undici:12502:13
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async eval (webpack-internal:///(rsc)/./node_modules/@prisma/extension-optimize/dist/index.js:1:12256)
    at async Ze (webpack-internal:///(rsc)/./node_modules/@prisma/extension-optimize/dist/index.js:1:14144)
    at async Q.onFlush (webpack-internal:///(rsc)/./node_modules/@prisma/extension-optimize/dist/index.js:1:18725)
    at async Q.forceFlush (webpack-internal:///(rsc)/./node_modules/@prisma/extension-optimize/dist/index.js:1:15626) {
  [cause]: ConnectTimeoutError: Connect Timeout Error (attempted addresses: 2a03:2880:f12a:83:face:b00c:0:25de:443, 157.240.18.18:443)
      at onConnectTimeout (node:internal/deps/undici/undici:6635:28)
      at node:internal/deps/undici/undici:6587:50
      at Immediate._onImmediate (node:internal/deps/undici/undici:6617:37)
      at process.processImmediate (node:internal/timers:478:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17) {
    code: 'UND_ERR_CONNECT_TIMEOUT'
  }
}

```

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!;

**Reply by jcyuan** [reply]
no luck, with `^1.1.4` still the same timeout error

**Comment by jcyuan** [comment]
update: with latest prisma/client 6.3.1 + @prisma/extension-optimize 1.1.4

same error

and new errors:

`Critical dependency: require function is used in a way in which dependencies cannot be statically extracted` while doing trace importing
